xicam/gui/bluesky/search.py

```diff
@@ -407,24 +408,80 @@ def __init__(self, *args, **kwargs):
         self.horizontalHeader().setSectionResizeMode(QHeaderView.ResizeToContents)
         self.setAlternatingRowColors(True)
 
+    def hide_hidden_columns(self):
+        hidden_columns = QSettings().value("catalog.columns.hidden")
+        if hidden_columns is None:
+            return
+        header = self.horizontalHeader()
+        current_column_names = [str(self.model().headerData(i, Qt.Horizontal)) for i in range(header.count())]
+        for column_name in hidden_columns:
```

Comment:
Since `hidden_columns` is a `set` it might be natural to do a set intersection here instead of nested loops.

Reply:
:+1: